website/for_teachers.py

```diff
@@ -208,6 +208,26 @@ def get_class(self, user, class_id):
             survey_later=survey_later,
         )
 
+    @route("/class/<class_id>/preview", methods=["GET"])
+    @requires_login
+    def preview_class_as_teacher(self, user, class_id):
+        if not is_teacher(user) and not is_admin(user):
+            return utils.error_page(error=403, ui_message=gettext("retrieve_class_error"))
+        Class = self.db.get_class(class_id)
+        if not Class or (not utils.can_edit_class(user, Class) and not is_admin(user)):
+            return utils.error_page(error=404, ui_message=gettext("no_such_class"))
+        session["preview_class"] = Class["id"]
```

Comment:
perhaps we should save a dict with `id` and `class_name` class name instead.

Reply:
Agreed